analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -66,5 +68,34 @@ private ProgramState()
             other is not null
             && other.OperationValue.DictionaryEquals(OperationValue)
             && other.SymbolValue.DictionaryEquals(SymbolValue);
+
+        public override string ToString()
+        {
+            if (Equals(Empty))
+            {
+                return "Empty";
+            }
+            else
+            {
+                var sb = new StringBuilder();
+                if (SymbolValue.Any())
+                {
+                    sb.AppendLine("Symbols:");
+                    foreach (var kvp in SymbolValue.OrderBy(x => x.Key.ToString()))
+                    {
+                        sb.Append(kvp.Key).Append(": ").AppendLine(kvp.Value?.ToString() ?? "<null>");
+                    }
+                }
+                if (OperationValue.Any())
+                {
+                    sb.AppendLine("Operations:");
+                    foreach (var kvp in OperationValue.Select(x => new KeyValuePair<string, string>(x.Key.Serialize(), x.Value?.ToString() ?? "<null>")).OrderBy(x => x.Key))
+                    {
+                        sb.Append(kvp.Key).Append(": ").AppendLine(kvp.Value);
+                    }
+                }
+                return sb.ToString();
+            }
```

Comment:
I think we should extract 2 methods: GetSymbols, GetOperations and then transform this full method into an expression.